### ISSUE TITLE
server return 304 code

### ISSUE THREAD

**Issue by seyaobey** [TypedProject/ts-express-decorators#550]
## Information

- **Version:** 5.2.0
- **Type:** Issue

## Description
I have built a fairly simple application (basically I am evaluating this library) with just one controller and basic crud operations + a bunch of routes agains a mongodb database. My controller has 7 routes. And all works: meaning they correctly return the result of their call to client. 

Now I added a 8th route returning the result of an aggregate function. And this particular route returns a `304` code. I can not understand why. Now I cannot upload the whole project here for you to reproduce this issue. I will just display the code which present the problem

## 1. MyController

```typescript
import {Controller, Get} from "@tsed/common";
import { Summary } from "@tsed/swagger";

@Controller('/mycontroller')
export class MyController {

  // other crud routes...

  @Summary("route description")
  @Get("/listing")
  async getGroupListing() {
    return await this.dataService.getGroupListing();
  }
}
```
The `dataService` object has been correctly injected in the constructor. I show here the content of its `getGroupListing` method:

```typescript
  async getGroupListing() {
    const data = await this.model.aggregate([ // model is an injected mongoose model
      {
        $lookup: {
          from: "likTemplates",
          localField: "_id",
          foreignField: "templateid",
          as: "template"
        }
      }
    ]);

    return data;
  }
```

Putting a breakpoint in debug mode, I can see that the query returns some data. But I get a 304 when I call this route (no result when using swagger or postman). 

<img width="1305" alt="code_304" src="https://user-images.githubusercontent.com/2738569/57562831-43b50d80-738e-11e9-8910-354ac7d49b6e.png">

Please can you point me to some direction as to what could this problem? Is there a limit to the number of routes I can create in a controller? 

Your help and guidance will be much appreciated. I have spent 3 days trying to figure out what could be the problem and looking across internet. I would like to continue using this library because so far, it is the best I found dealing with typescript decorators for rest apis. 

Here is my server settings:

```typescript
@ServerSettings({
  port: process.env.PORT || 8080,
  rootDir,
  acceptMimes: ["application/json"],
  mount: {
    "/api": "${rootDir}/controllers/**/*.ts"
  },
  componentsScan: [`./services/**/**.ts`, "./middlewares/**/**.ts"],
  swagger: [
    {
      path: "/api-docs"
    }
  ]
})
export class Server extends ServerLoader {
  public $onMountingMiddlewares(): void | Promise<any> {
    this.use(GlobalAcceptMimesMiddleware)
      .use(cookieParser())
      .use(compress({}))
      .use(methodOverride())
      .use(bodyParser.json())
      .use(cors())
      .use(
        bodyParser.urlencoded({
          extended: true
        })
      )
      .use(boom());

    return null;
  }

  public async $onInit(): Promise<any> {
    // start mongoose
    await mongoose.connect(db_connection, {
      auth: {
        user: "__user_name__",
        password: "__password__"
      },
      useNewUrlParser: true
    });
    console.log("Mongodb started");
  }

  public $onReady() {
    console.log("Server started");
  }

  public $onServerInitError(err) {
    console.error(err);
  }
```

## Acceptance criteria

- [ ] Criteria 1

**Comment by Romakita** [comment]
Hello @seyaobey 
Your error is a bit strange because nothing in the Ts.ED framework is implemented to send automatically a 304 status code. 
There is not limit about the route number for each controller but the routes order is very importante (in Express). 

Normally, if you have your route in routes list displayed in the terminal when you start your server, that mean your route is available. A quick test can help me about your bug. 

```typescript
import {Controller, Get} from "@tsed/common";
import { Summary } from "@tsed/swagger";

@Controller('/mycontroller')
export class MyController {

  // other crud routes...

  @Summary("route description")
  @Get("/listing")
  async getGroupListing() {
    console.log('====> OK')
    const result = await this.dataService.getGroupListing();
    console.log('====> OK', result)
    return result
  } 
}
```
Can you tell me if the log appear when your send a request ? If not, we can continue this discussion on gitter https://gitter.im/Tsed-io/community?source=orgpage . Maybe a direct access on your project could save us time (if it's possible). 

Additionally, I see you use mongoose. Ts.ED provide a package to manage mongoose with decorators and simplify the configuration:
http://tsed.io/tutorials/mongoose.html#declaring-a-mongoose-object-schema-or-model

See you
Romain

**Comment by seyaobey** [comment]
Thanks @Romakita for coming back to me. The log appears in the console when the request is sent.
I could not figure out what's going on. I just created a new clean project from scratch and transfer my code into the new project. Now it is working. Maybe something has been corrupted somewhere ( in some in my project? (`/node_modules` files?). Thanks for your help though!

**Comment by lcatania** [comment]
I get the same Error and just tried to get a fresh copy of the example. But the example is broken it doesn't find the @Returns decorator that is on the endpoints . Maybe this will fix this weird behavior. I saw that in v6 deprecated Decorators will be removed . @Romakita Do you know when v6 will be out.

Thanks guys

**Comment by Romakita** [comment]
Hello @lcatania  
Returns decorator should be available from @tsed/common package (since v5.62.3).

V6 is under alpha channel (https://v6.tsed.io/). I work actively on v6 to give RC version on september.

See you
Romain
